### PR TITLE
[#2184] improve(iceberg-rest): Improved test coverage for IcebergMetricsFormatter

### PR DESCRIPTION
**Pull request by yaoderek**
1. [#2184 ] test: Added test coverage for IcebergMetricsFormatter

### What changes were proposed in this pull request?

Adding test cases to improve coverage of IcebergMetricsFormatter

### Why are the changes needed?

Currently very low coverage for this class. 

Fix: #2184

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Running IntelliJ with coverage. Code coverage Number before and after are attached.

**After the Fix**
after

** Before the Fix**
before change

### CODE REVIEW

**Comment by FANNG1** [comment]
@yaoderek , thanks for your PR, as you could see , `IcebergMetricsFormatter` is used to generate json string for the log, seems no need to add test to improve the test coverage.  cc @justinmclean

**Comment by justinmclean** [comment]
I don't see any harm in adding the test.

File: iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/metrics/TestIcebergMetricsFormatter.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        .sequenceNumber(1)
+        .operation("select")
+        .commitMetrics(commitMetricsResult)
+        .build();
+  }
+
+  private static IcebergMetricsFormatter formatter;
+  private static MetricsReport report;
+  private static final String expectedString =
+      "{\"table-name\":\"a\",\"snapshot-id\":1,\"sequence-number\":1,\"operation\":\"select\","
```
Comment (FANNG1) [inline iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/metrics/TestIcebergMetricsFormatter.java:47]: the test result relies on the content of `ImmutableCommitMetricsResult` which is out of our control and hard to maintain if the upstream adds some fields to it.

Comment (justinmclean) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/metrics/TestIcebergMetricsFormatter.java:47]: Perhaps just check for the values we set?

Comment (FANNG1) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/metrics/TestIcebergMetricsFormatter.java:47]: yes, this is an acceptable solution.

Comment (xiaoyuyao) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/metrics/TestIcebergMetricsFormatter.java]: Good point. I posted an update to the PR.

Comment (yaoderek) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/metrics/TestIcebergMetricsFormatter.java]: Good point. I posted an update.

Comment (justinmclean) [reply iceberg/iceberg-rest-server/src/test/java/org/apache/gravitino/iceberg/service/metrics/TestIcebergMetricsFormatter.java]: LGTM, thanks for your contribution

**Review by FANNG1 — APPROVED** [review]

**Pull request merged**